modules/tinymce/src/core/main/ts/api/util/EventDispatcher.ts

```diff
@@ -183,6 +184,13 @@ class EventDispatcher<T> {
       for (let i = 0, l = handlers.length; i < l; i++) {
         const callback = handlers[i];
 
+        // The handler was removed by an earlier handler in this loop so skip it. This is needed as
+        // the `handlers` array is not a "live" list, so when the handler is removed while we are
```

Comment:
I'm not a fan of saying that this isn't a "live" list, let's find a better alternative.

Firstly, if someone _adds_ a new event handler while we're in the middle of this loop, it will be included in this loop as though it were a "live" list, because we just call `.push` on the same underlying list.

Secondly, I feel like we could be setting ourselves up for trouble by not stressing how important it is that this section of the code, and the matching section of the `off` function 120 lines below, are always kept in sync. I can see myself:

1. looking at a comment explaining how this list isn't "live"
2. looking at how we pulled a (super-duper-shallow) copy of the handlers list out of the big bindings object a few lines earlier
3. thinking "that (super-duper-shallow) copy of the handlers list stops it from being live", and then
4. messing something up if I tried to change this code (eg mutating on `this.bindings[name]` in another function and expecting it not to interfere with this loop)

The best suggestion I can think of right now is something like:

> This is needed as we remove callbacks from the handlers list by creating a new list and updating the reference in `this.bindings[name]`, which won't affect our `handlers` array.

But I'm not super attached to that wording, as long as the message gets through (that it's linked super closely to the removal process)

Reply:
As discussed I've made the handlers a clone instead to make it clearer and done some other minor rework.